pennylane/transforms/defer_measurements.py

```diff
@@ -101,9 +106,299 @@ def null_postprocessing(results):
+                data[idx] = value
+                op = jax.tree_util.tree_unflatten(struct, data)
+                qml.ctrl(op, mv.wires, control_values=branch)
+
+        def interpret_operation(self, op: "pennylane.operation.Operator"):
+            """Interpret a PennyLane operation instance.
+
+            Args:
+                op (Operator): a pennylane operator instance
+
+            Returns:
+                Any
+
+            This method is only called when the operator's output is a dropped variable,
+            so the output will not affect later equations in the circuit.
+
+            See also: :meth:`~.interpret_operation_eqn`.
+
+            """
+            data, struct = jax.tree_util.tree_flatten(op)
+
+            idx = -1
+            for i, d in enumerate(data):
+                if isinstance(d, MeasurementValue):
+                    idx = i
+                    break
+            if idx != -1:
+                return self.interpret_dynamic_operation(data, struct, idx)
+
+            return jax.tree_util.tree_unflatten(struct, data)
+
+        def interpret_measurement(self, measurement: "qml.measurement.MeasurementProcess"):
+            """Interpret a measurement process instance.
+
+            Args:
+                measurement (MeasurementProcess): a measurement instance.
+
+            See also :meth:`~.interpret_measurement_eqn`.
+
+            """
+            if measurement.mv is not None:
+                kwargs = {"wires": measurement.wires, "eigvals": measurement.eigvals()}
+                if isinstance(measurement, CountsMP):
+                    kwargs["all_outcomes"] = measurement.all_outcomes
+
+                measurement = type(measurement)(**kwargs)
+
+            return super().interpret_measurement(measurement)
+
+        def resolve_mcm_values(
+            self, eqn: "jax.core.JaxprEqn", invals: Sequence[Union[MeasurementValue, Number]]
+        ) -> MeasurementValue:
+            """Create a ``MeasurementValue`` that captures all classical processing of the
+            input ``eqn`` in its ``processing_fn``.
+
+            Args:
+                eqn (jax.core.JaxprEqn): Jaxpr equation containing the primitive to apply
+                invals (Sequence[Union[MeasurementValue, Number]]): Inputs to the equation
+
+            Returns:
+                MeasurementValue: ``MeasurementValue`` containing classical processing information
+                for applying the input equation to mid-circuit measurement outcomes.
+            """
+            # pylint: disable=protected-access,unnecessary-lambda
+            assert len(invals) <= 2
+
+            if len(invals) == 1:
+                meas_val = invals[0]
+                processing_fn = lambda x: eqn.primitive.bind(x, **eqn.params)
+                return meas_val._apply(processing_fn)
+
+            if all(isinstance(inval, MeasurementValue) for inval in invals):
+                processing_fn = lambda *x: eqn.primitive.bind(*x, **eqn.params)
```

Comment:
```suggestion
                def processing_fn(*x):
                    return eqn.primitive.bind(*x, **eqn.params)
```
no more unnecessary lambda.